src/sage/graphs/matching_covered_graph.py

```diff
@@ -0,0 +1,534 @@
+        Traceback (most recent call last):
+        ...
+        ValueError: the graph is not connected
+
+    Make sure that self-loops are not allowed for a matching covered graph::
+
+        sage: P = graphs.PetersenGraph()
+        sage: G = MatchingCoveredGraph(P)
+        sage: G.allows_loops()
+        False
+        sage: G.allow_loops(True)
+        Traceback (most recent call last):
+        ...
+        ValueError: loops are not allowed in matching covered graphs
+        sage: G.add_edge(0, 0)
+        Traceback (most recent call last):
+        ...
+        ValueError: cannot add edge from 0 to 0 in graph without loops
+        sage: H = MatchingCoveredGraph(P, loops=True)
+        Traceback (most recent call last):
+        ...
+        ValueError: loops are not allowed in matching covered graphs
+
+    Make sure that multiple edges are allowed for a matching covered graph (by
+    default it is off and can be modified to be allowed)::
+
+        sage: P = graphs.PetersenGraph()
+        sage: G = MatchingCoveredGraph(P)
+        sage: G
+        Matching covered petersen graph: graph on 10 vertices
+        sage: G.allows_multiple_edges()
+        False
+        sage: G.size()
+        15
+        sage: G.allow_multiple_edges(True)
+        sage: G.allows_multiple_edges()
+        True
+        sage: G.add_edge(next(P.edge_iterator()))
+        sage: G.size()
+        16
+        sage: G
+        Matching covered petersen graph: multi-graph on 10 vertices
+        sage: H = MatchingCoveredGraph(P, multiedges=True)
+        sage: H.allows_multiple_edges()
+        True
+        sage: H.add_edge(next(P.edge_iterator()))
+        sage: H.size()
+        16
+        sage: H
+        Matching covered petersen graph: multi-graph on 10 vertices
+
+    Providing with a connected nontrivial graph free of self-loops that is
+    not matching covered::
+
+        sage: G = graphs.CompleteGraph(11)
+        sage: H = MatchingCoveredGraph(G)
+        Traceback (most recent call last):
+        ...
+        ValueError: input graph is not matching covered
+        sage: G = Graph({0: [1, 6, 11], 1: [2, 4], 2: [3, 5], 3: [4, 5],
+        ....:            4: [5], 6: [7, 9], 7: [8, 10], 8: [9, 10], 9: [10],
+        ....:            11: [12, 14], 12: [13, 15], 13: [14, 15], 14: [15]})
+        sage: H = MatchingCoveredGraph(G)
+        Traceback (most recent call last):
+        ...
+        ValueError: input graph is not matching covered
+        sage: # needs networkx
+        sage: import networkx
+        sage: G = Graph(networkx.complete_bipartite_graph(2, 12))
+        sage: H = MatchingCoveredGraph(G)
+        Traceback (most recent call last):
+        ...
+        ValueError: input graph is not matching covered
+        sage: G = Graph('F~~~w')
+        sage: H = MatchingCoveredGraph(G)
+        Traceback (most recent call last):
+        ...
+        ValueError: input graph is not matching covered
+        sage: # needs sage.modules
```

Comment:
add an empty line before `sage: # needs sage.modules`. Otherwise, all the tests after will also need. networkx`.